### PR TITLE
Update MSstats integration

### PR DESCRIPTION
**Pull request by wfondrie**
The PR adds better MSstats integration and updates the default EncyclopeDIA parameters to match the EncyclopeDIA GUI defaults.

### CODE REVIEW

**Comment by wfondrie** [comment]
@cia23, I've finally got this working correctly! Please take a look when you're able so I can get the final analyses going.

File: bin/msstats.R [hunk 1]
```diff
@@ -1,76 +1,132 @@
...
            FragmentIon,
            ProductCharge)
 
-  df
+  return(df)
+}
+
+
+# Parse the annotation file.
+annotate <- function(peptide_df, annot_csv) {
```
Comment (cia23) [inline bin/msstats.R:49]: Where is this annotation file coming from? Is it the input file with the added columns that we discussed earlier this week?

File: main.nf [hunk 2]
```diff
@@ -87,16 +103,30 @@ workflow {
     | set { quant_files }
 
     // Analyze the quantitative runs with EncyclopeDIA.
-    // The output has three channels:
+    // The output has two channels:
```
Comment (cia23) [inline main.nf:106]: Why did this change from 3 down to 2?

File: tests/unit_tests/test_msstats.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+def test_reports(msstats_input, script):
+    """Test without reports"""
+    peptide_file, annot_file, contrast_file = msstats_input
+    args = [
+        script,
+        peptide_file,
+        annot_file,
+        contrast_file,
+        "equalizeMedians",
```
Comment (cia23) [inline tests/unit_tests/test_msstats.py:26]: I assume equalizeMedians is not the only thing we can do with msstats. What are the other options with msstats? Have we tested anything else?

**Review by cia23 — APPROVED** [review]
The changes look really good! I had a couple clarifying questions, but the code looks good.

File: bin/msstats.R (hunk 1 above)
Comment (wfondrie) [reply bin/msstats.R:49]: Yes, exactly! The input file can optionally contain `condition` and `bioreplicate` columns, which are used here. I'm prepping another PR that greatly improves the documentation, which I'll wrap up once we have the paper together.

File: tests/unit_tests/test_msstats.py (hunk 3 above)
Comment (wfondrie) [reply tests/unit_tests/test_msstats.py:26]: You're absolutely correct. My knowledge might be dated, but MSstats offers 3 options: no normalization, median normalization, and quantile normalization. Each has their place, but generally the more transformations you need to do, the more signal is lost; thus, the general advice is use as much normalization as you need, but no more.

As for testing them, we've not really explored what MSstats can do for us. It is worth keeping in mind that EncyclopeDIA already tries to normalize peptide quants to the total ion current (TIC), which should be proportional to the amount of protein loaded.

**Pull request merged**